tuf/formats.py

```diff
@@ -789,20 +793,64 @@ def make_fileinfo(length, hashes, version=None, custom=None):
     will raise a 'securesystemslib.exceptions.FormatError' exception.
 
   <Returns>
-    A dictionary conformant to 'FILEINFO_SCHEMA', representing the file
+    A dictionary conformant to 'TARGETS_FILEINFO_SCHEMA', representing the file
     information of a metadata or target file.
   """
 
   fileinfo = {'length' : length, 'hashes' : hashes}
 
-  if version is not None:
-    fileinfo['version'] = version
-
   if custom is not None:
     fileinfo['custom'] = custom
 
   # Raise 'securesystemslib.exceptions.FormatError' if the check fails.
-  FILEINFO_SCHEMA.check_match(fileinfo)
+  TARGETS_FILEINFO_SCHEMA.check_match(fileinfo)
+
+  return fileinfo
+
+
+
+def make_metadata_fileinfo(version, length=None, hashes=None):
+  """
+  <Purpose>
+    Create a dictionary conformant to 'METADATA_FILEINFO_SCHEMA'.
+    This dict describes both metadata and target files.
+
+  <Arguments>
+    version:
+      An integer representing the version of the file.
+
+    length:
+      An optional integer representing the size of the file.
+
+    hashes:
+      An optional dict of hashes in 'HASHDICT_SCHEMA' format, which has the form:
+       {'sha256': 123df8a9b12, 'sha512': 324324dfc121, ...}
+
+
+  <Exceptions>
+    securesystemslib.exceptions.FormatError, if the 'METADATA_FILEINFO_SCHEMA' to be
+    returned does not have the correct format.
+
+  <Side Effects>
+    If any of the arguments are incorrectly formatted, the dict
+    returned will be checked for formatting errors, and if found,
+    will raise a 'securesystemslib.exceptions.FormatError' exception.
```

Comment:
I appreciate the diligence here, but I think we don't need this Side Effects section in this case. I'd expect this part of the docstring to describe side-effects that weren't obvious from the rest of the docstring. We already explain the Exception in the section above.

Reply:
Agree. I will remove the side effect section from the `make_targets_fileinfo` as well.